src/string-builder.js

```diff
@@ -0,0 +1,27 @@
+module.exports = StringBuilder;
+
+function StringBuilder(){
+    this.buffer = [];
+}
+
+StringBuilder.prototype.cat = function(){
+    var buffer = this.buffer;
+    concat(...arguments);
+    
+    function concat(a, ...rest) {
```

Comment:
It feels like you are abusing of recursion, what about using the old and good friend `for(;;)`?